crates/burn-jit/src/fusion/on_write/trace.rs

```diff
@@ -273,9 +274,9 @@ impl FuseOnWriteTrace {
                     }
                 }
 
-                // We encode bool tensors as u32.
+                // We encode bool tensors as `B`.
                 let dtype = match tensor_global.dtype {
-                    DType::Bool => DType::U32,
+                    DType::Bool => B::dtype(),
```

Comment:
Would it be better to pass the `bool_dtype` in the constructor instead of keeping the generic argument?

Reply:
I think the generic is simpler for now